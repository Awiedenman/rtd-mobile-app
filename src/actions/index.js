import { userInfo } from "os";

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});
   
export const hasErrored = (bool) => ({
  type: 'HAS_ERRORED',
  hasErrored: bool
});

export const storeScheduleData = (schedule) => ({
  type: 'STORE_SCHEDULE_DATA',
  schedule
});

export const storeStartSearch = (startAddress) => ({
  type: 'STORE_START_SEARCH',
  startAddress
});

export const storeEndSearch = (endAddress) => ({
  type: 'STORE_END_SEARCH',
  endAddress
});

export const storeMockFavRoutes = (mockFavRoutes) => ({
  type: 'STORE_MOCK_FAV_ROUTES',
  mockFavRoutes
});

export const signUpUser = (userInfo) => ({
  type: 'SIGN_UP_USER',
  userInfo
});