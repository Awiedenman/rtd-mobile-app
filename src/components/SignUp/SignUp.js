import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';

import { auth } from '../../firebase';

import * as routes from '../../constants/routes';

const SignUpPage = ({ history }) => {
  return (
    <div>
      <h1>Sign Up</h1>
      <SignUpForm history={history} />
    </div>
  );
};

const INITIAL_STATE = {
  username: '',
  email: '',
  passwordOne: '',
  passwordTwo: '',
  error: null
};

const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value
});


class SignUpForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      ...INITIAL_STATE
    };
  }

    onSubmit = (event) => {
      const {
        username,
        email,
        passwordOne
      } = this.state;

      const { history } = this.props;
  
      auth.doCreateUserWithEmailAndPassword(email, passwordOne)
        .then(authUser => {
          this.setState({ ...INITIAL_STATE });
          history.push(routes.HOME);
        })
        .catch(error => {
          this.setState(byPropKey('error', error));
        });
  
      event.preventDefault();
    }

    handleChange = (event) => {
      const { name, value } = event.target;
      this.setState({
        [name]: value
      });
    };

    render() {
      const {
        username,
        email,
        passwordOne,
        passwordTwo,
        error
      } = this.state;

      const isInvalid =
      passwordOne !== passwordTwo ||
      passwordOne === '' ||
      email === '' ||
      username === '';

      return (
        <form onSubmit={this.onSubmit}>
          <input
            name="username"
            value={username}
            onChange={this.handleChange}
            type="text"
            placeholder="Full Name"
          />
          <input
            name="email"
            value={email}
            onChange={this.handleChange}
            type="email"
            placeholder="Email Address"
          />
          <input
            name="passwordOne"
            value={passwordOne}
            onChange={this.handleChange}
            type="password"
            placeholder="Password"
          />
          <input
            name="passwordTwo"
            value={passwordTwo}
            onChange={this.handleChange}
            type="password"
            placeholder="Confirm Password"
          />
          <button 
            type="submit"
            disabled={isInvalid}
          >
          Sign Up
          </button>

          { error && <p>{error.message}</p> }
        </form>
      );
    }
}

const SignUpLink = () => {
  return (
    <p>
    Don't have an account?
      {' '}
      <Link to={routes.SIGN_UP}>Sign Up</Link>
    </p>
  );

};

export default withRouter(SignUpPage);

export {
  SignUpForm,
  SignUpLink
};