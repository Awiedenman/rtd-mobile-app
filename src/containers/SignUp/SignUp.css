.sign-up-page {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.sign-up-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: white;
  margin-top: 10px;
  width: 95%;
  box-shadow: 1px 1px 2px 2px rgba(0, 0, 0, 0.5);
  padding-bottom: 10px;
}

.sign-up-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
}

.sign-up-title {
  margin: 20px;
}

.sign-up-input {
  height: 40px;
  width: 90%;
  border: 1px solid gray;
  border-bottom: none;
}

#password-two {
  border-bottom: 1px solid gray;
}