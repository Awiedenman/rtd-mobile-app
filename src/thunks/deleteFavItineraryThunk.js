import { 
  isLoading, 
  hasErrored, 
  deleteFavorite 
} from '../actions';

const deleteFavItineraryThunk = (fetchObj) => {
  const {
    url,
    options
  } = fetchObj;
  return (dispatch) => {
    dispatch(isLoading(true));
    fetch(url, options)
      .then(response => {
        if (!response.ok) {
          throw Error();
        }
        dispatch(isLoading(false));
        return response;
      })
      .then(response => response.json())
      .then(favorite => {
        dispatch(deleteFavorite(favorite));
      })
      .catch(() => dispatch(hasErrored(true)));
  };
};

export default deleteFavItineraryThunk;
