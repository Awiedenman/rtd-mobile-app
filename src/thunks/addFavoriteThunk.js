import { isLoading, hasErrored, addFavorite,  } from '../actions';

export const addFavoriteThunk = (fetchObj) => {
  const {
    url,
    options,
  } = fetchObj;
  return (dispatch) => {
    dispatch(isLoading('favorite'));
    fetch(url, options)
      .then(response => {
        if (!response.ok) {
          throw Error();
        }
        dispatch(isLoading(null));
        return response;
      })
      .then(response => response.json())
      .then(favorite => dispatch(addFavorite(favorite)))
      .catch(() => dispatch(hasErrored(true)));
  };
};

export default addFavoriteThunk;