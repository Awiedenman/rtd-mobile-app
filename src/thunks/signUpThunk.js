import { isLoading, hasErrored, signUpUser } from '../actions';

const signUpThunk = (fetchObj) => {
  console.log('click');
  const {
    url,
    options
  } = fetchObj;
  return (dispatch) => {
    dispatch(isLoading(true));
    fetch(url, options)
      .then(response => {
        if (!response.ok) {
          throw Error();
        }
        dispatch(isLoading(false));
        return response;
      })
      .then(response => response.json())
      .then(userInfo => {
        dispatch(signUpUser(userInfo));
      })
      .catch(() => dispatch(hasErrored(true)));
  };
};

export default signUpThunk;