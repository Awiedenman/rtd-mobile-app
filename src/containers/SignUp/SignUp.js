import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import signUpThunk from '../../thunks/signUpThunk';
import * as routes from '../../constants/routes';
import { auth } from '../../firebase';
import PropTypes from 'prop-types';
import { SignInLink } from '../SignIn/SignIn';


class SignUpPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      passwordOne: '',
      passwordTwo: '',
      error: null
    };
  }

  createUser = (userInfo) => ({
    url: 'http://rtd-revamp-api.herokuapp.com/api/v1/users',
    options: {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        username: this.state.username,
        uid: userInfo.user.uid,
        email: userInfo.user.email
      })
    }
  })
  
  resetForm = () => {
    this.setState({
      username: '',
      email: '',
      passwordOne: '',
      passwordTwo: '',
      error: null
    });
  }

  onSubmit = (event) => {
    event.preventDefault();

    const {
      email,
      passwordOne
    } = this.state;
    
    const { history, signUpThunk } = this.props;
    
    auth.doCreateUserWithEmailAndPassword(email, passwordOne)
      .then(authUser => {
        const newUser = this.createUser(authUser);
        return signUpThunk(newUser);
      })
      .catch(error => {
        this.setState({error: error});
      });
    this.resetForm();
    history.push(routes.HOME);
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  render() {
    const {
      username,
      email,
      passwordOne,
      passwordTwo,
      error
    } = this.state;

    const isInvalid =
    passwordOne !== passwordTwo ||
    passwordOne === '' ||
    email === '' ||
    username === '';

    return (
      <div className='sign-up-container'>
        <form onSubmit={this.onSubmit}>
          <input
            name='username'
            value={username}
            onChange={this.handleChange}
            type='text'
            placeholder='Full Name'
          />
          <input
            name='email'
            value={email}
            onChange={this.handleChange}
            type='email'
            placeholder='Email Address'
          />
          <input
            name='passwordOne'
            value={passwordOne}
            onChange={this.handleChange}
            type='password'
            placeholder='Password'
          />
          <input
            name='passwordTwo'
            value={passwordTwo}
            onChange={this.handleChange}
            type='password'
            placeholder='Confirm Password'
          />
          <button 
            type='submit'
            disabled={isInvalid}
          >
        Sign Up
          </button>

          { error && <p>{error.message}</p> }
        </form>
        <SignInLink />
      </div>
    );
  }
}

const SignUpLink = () => {
  return (
    <p>
    Don't have an account?
      {' '}
      <Link to={routes.SIGN_UP}>Sign Up</Link>
    </p>
  );

};

export const mapDispatchToProps = dispatch => ({
  signUpThunk: (userInfo) => dispatch(signUpThunk(userInfo))
});

export default withRouter(connect(null, mapDispatchToProps)(SignUpPage));

export {
  SignUpLink
};

SignUpPage.propTypes = {
  signUpThunk: PropTypes.func,
  history: PropTypes.object
};