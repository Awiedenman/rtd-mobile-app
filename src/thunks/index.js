import postItineraryThunk from './postItineraryThunk';
import signUpThunk from './signUpThunk';


export default {
  postItineraryThunk,
  signUpThunk
};