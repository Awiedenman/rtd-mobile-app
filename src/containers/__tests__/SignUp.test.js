import React from "react";
import { shallow } from "enzyme";
import { SignUpPage, mapDispatchToProps } from "../SignUp/SignUp";
import * as actions from "../../actions";
import { signUpThunk } from "../../thunks/signUpThunk";
import { createSecureServer } from "http2";

describe("SignUpPage", () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      signUpThunk: jest.fn(),
      history: { push: jest.fn() }
    };

    wrapper = shallow(<SignUpPage {...mockProps} />);
  });

  it("should match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("CreateUser", () => {
    it.skip("should have an initial state of no user info", () => {
      const userInfo = {
        user: {
          uid: "1",
          email: "mock@email.com"
        }
      };

      const expected = {
        url: "https://rtd-revamp-api.herokuapp.com/api/v1/users",
        options: {
          method: "POST",
          headers: {
            "Content-Type": "application/json"
          },
          body: JSON.stringify({
            username: "",
            uid: "1",
            email: "mock@email.com"
          })
        }
      };

      const createUser = wrapper.instance().createUser(userInfo);

      expect(createUser).toEqual(expected);
    });
  });

  describe("resetForm", () => {
    it("reset the state", () => {
      const mockState = {
        username: "Austin",
        email: "austin@aol.com",
        passwordOne: "password1",
        passwordTwo: "password2",
        error: null
      };

      wrapper.setState(mockState);

      const expected = {
        username: "",
        email: "",
        passwordOne: "",
        passwordTwo: "",
        error: null
      };

      wrapper.instance().resetForm();

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe("onSubmit", () => {
    it("calls preventDefault", () => {
      const mockEvent = { preventDefault: jest.fn() };

      wrapper.instance().onSubmit(mockEvent);

      expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
    });

    it.skip("calls resetForm", () => {
      const mockEvent = { preventDefault: jest.fn() };
      const resetForm = (wrapper.instance().resetForm = jest.fn());

      wrapper.instance().onSubmit(mockEvent);

      expect(resetForm).toHaveBeenCalledTimes(1);
    });

    it.skip("calls history.push", () => {
      const mockEvent = { preventDefault: jest.fn() };

      wrapper.instance().onSubmit(mockEvent);

      expect(mockProps.history.push).toHaveBeenCalledTimes(1);
    });
  });

  describe("handleChange", () => {
    it("sets state of name and value", () => {
      const mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: "username",
          value: "Austin"
        }
      };

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state().username).toEqual("Austin");
    });
  });

  describe("mapDispatchToProps", () => {
    it.skip("calls dispatch with the correct params on signUpThunk", () => {
      const mockDispatch = jest.fn();
      const userInfo = {
        username: "mockUserName",
        id: "1",
        email: "mock@email.com"
      };
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = signUpThunk(userInfo);
      mappedProps.signUpThunk(userInfo);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});
