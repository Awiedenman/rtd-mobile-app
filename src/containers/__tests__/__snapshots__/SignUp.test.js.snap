// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage should match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpPage
    history={
        Object {
            "push": [Function],
          }
    }
    signUpThunk={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    name="username"
                    onChange={[Function]}
                    placeholder="Full Name"
                    type="text"
                    value=""
          />
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="email"
                    value=""
          />
          <input
                    name="passwordOne"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <input
                    name="passwordTwo"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign Up
          </button>
</form>,
        <Unknown />,
      ],
      "className": "sign-up-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="username"
              onChange={[Function]}
              placeholder="Full Name"
              type="text"
              value=""
/>,
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="email"
              value=""
/>,
            <input
              name="passwordOne"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <input
              name="passwordTwo"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign Up
</button>,
            null,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "username",
              "onChange": [Function],
              "placeholder": "Full Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordOne",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordTwo",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
          null,
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="username"
                        onChange={[Function]}
                        placeholder="Full Name"
                        type="text"
                        value=""
            />
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Email Address"
                        type="email"
                        value=""
            />
            <input
                        name="passwordOne"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <input
                        name="passwordTwo"
                        onChange={[Function]}
                        placeholder="Confirm Password"
                        type="password"
                        value=""
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign Up
            </button>
</form>,
          <Unknown />,
        ],
        "className": "sign-up-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="username"
                onChange={[Function]}
                placeholder="Full Name"
                type="text"
                value=""
/>,
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="email"
                value=""
/>,
              <input
                name="passwordOne"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <input
                name="passwordTwo"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign Up
</button>,
              null,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "username",
                "onChange": [Function],
                "placeholder": "Full Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "passwordOne",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "passwordTwo",
                "onChange": [Function],
                "placeholder": "Confirm Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
            null,
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
